projects/eods-specification.md

```diff
@@ -0,0 +1,266 @@
+
+### Actively Validated Service” (AVS), as Delegator role selection
+
+The second part of the project consists of the conceptual design of a plug-and-play interface for future integration of protocol services and an MVP specification of the interface, as minimal expected deliverables. 
+
+Post ePBS, eODS unlocks the introduction of two distinct types of protocol services:
+
+  * Consensus (Finality) services - FFG type
+  
+  * Censorship Resistant services - AVS type
+  
+    Delegators could re-stake the Operator's *liability*, into CR services e.g. inclusion lists, committing to the provision of an “actively validated service” (AVS) and receiving rewards for good service provision. 
+
+Possible separation of protocol services post ePBS:
+
+![Protocol Services](https://hackmd.io/_uploads/rJpoIhJKR.png)
+
+#### Conceptual design of an interface for adding protocol light services:
+* General design principle
+    * Design constraints
+    * Identify light & heavy operators and other stakeholders
+    * Identify services
+        * Finality gadget 
+        * Censorship-resistance gadgets (e.g. Inclusion lists, Multiplicity gadgets)
+    * Economics of consensus services
+    * Economics of AVS 
+    * MVI of the different protocol services
+* Specification notes referencing:
+    * An adaptation of the slashing mechanism to account for partially slashable light services
+    * Liveness
+    * Protocol safety
+
+## Part 1 - eODS Specification
+**How will you implement your solutions? Give details and more technical information on the project.**
+
+### What are the minimum set of requirements for eODS design?
+- New entity in beacon state: Delegators as a new class, similar to validators but with different signature domain and participation flags indices
+- Add state Delegator index & balances
+- Mapping in-protocol Principal-Agent relationship by explicitating a way for Delegators to transfer ETH to Validators and introducing Validator *"liabilities"* towards Delegators
+- Allow for delegator triggered `0x01` withdrawals.
+
+### Specification overview
+
+A sketch of the proposed **execution layer** changes is included below:
+
+#### Staking-deposit-cli
+Stakers deposit ETH in the protocol, provided `amount` >= `MIN_DEPOSIT_AMOUNT`. During [deposit](###Deposit), a forked staking-deposit-cli will allow depositors to set a boolean `is_delegator` field to `True` or `False`, alongside the address of a smart contract (delegation contract) that, when called, outputs the target validator's pubkey and withdrawal credentials. The structure denoting the deposit operation on EL will also gain the two new fields, accordingly.
+
+#### Deposit contract 
+The deposit contract will gain the following arguments: `is_delegator`, and `delegation_contract`. The `DepositData` container will be extended accordingly.
```

Comment:
Are you referring to breaking serialization by adding new Container fields?
- if yes, then I reckon eODS will need switching to [StableContainer](https://eips.ethereum.org/EIPS/eip-7495) 

Otherwise, the events of a DELEGATION_CONTRACT could be parsed in-protocol the same way we parse DEPOSIT_CONTRACT events currently, but there would still be the need for the depositor making the deposit to opt in on being a Delegator and not entering Validator activation queue.

Reply:
Yes I was referring to the second thing, that you need to deploy a separate DELEGATION_CONTRACT